ws-client/src/helpers.rs

```diff
@@ -72,6 +72,27 @@ pub fn process_subscription_response(
 	}
 }
 
+/// Attempts to process an incoming notification
+///
+/// Returns Ok() if the response was successfully handled
+/// Returns Err() if there was no handler for the method
+pub fn process_notification(manager: &mut RequestManager, notif: JsonRpcNotifResponse<JsonValue>) -> Result<(), Error> {
+	match manager.as_notification_handler_mut(notif.method.to_owned()) {
+		Some(send_back_sink) => match send_back_sink.try_send(notif.params) {
+			Ok(()) => Ok(()),
+			Err(err) => {
+				log::error!("Error sending notification, dropping handler for {:?} error: {:?}", notif.method, err);
+				let _ = manager.remove_notification_handler(notif.method.to_owned());
```

Comment:
@niklasad1 So we should remove the handler her / drop it in this scope?

Reply:
yepp, that should be it.

if you could just port this [test](https://github.com/paritytech/jsonrpsee/blob/master/tests/tests/integration_tests.rs#L139-#L166) to the register_notification that would be great, then it should be good

feel free to split it into two tests if you want.

Reply:
done